crypto/params_from_text.c

```diff
@@ -197,6 +199,111 @@ static int construct_from_text(OSSL_PARAM *to, const OSSL_PARAM *paramdef,
     return 1;
 }
 
+/**
+ * OSSL_PARAM_print_to_bio - Print OSSL_PARAM array to a bio 
+ *
+ * @p:        Array of OSSL_PARAM structures containing keys and values.
+ * @bio:      Pointer to bio where the formatted output will be written.
+ * @print_values: If non-zero, prints both keys and values. If zero, only keys
+ *                are printed.
+ *
+ * This function iterates through the given array of OSSL_PARAM structures,
+ * printing each key to an in-memory buffer, and optionally printing its
+ * value based on the provided data type. Supported types include integers,
+ * strings, octet strings, and real numbers.
+ *
+ * Return:    1 on success, 0 on failure.
+ */
+int OSSL_PARAM_print_to_bio(const OSSL_PARAM p[], BIO *bio, int print_values)
```

Comment:
For some reason I prefer `*p` to `p[]` - it is also more consistent with other functions working with OSSL_PARAM arrays. Also in the man page you have `*p`.